npm/angular/src/mount.ts

```diff
@@ -0,0 +1,211 @@
+   */
+  component: T
+};
+
+/**
+ * Bootstraps the TestModuleMetaData passed to the TestBed
+ *
+ * @param {Type<T>} component Angular component being mounted
+ * @param {TestBedConfig} config TestBed configuration passed into the mount function
+ * @returns {TestBedConfig} TestBedConfig
+ */
+function bootstrapModule<T extends object> (
+  component: Type<T>,
+  config: TestBedConfig<T>,
+): TestBedConfig<T> {
+  const { inputs, ...testModuleMetaData } = config
+
+  if (!testModuleMetaData.declarations) {
+    testModuleMetaData.declarations = []
+  }
+
+  if (!testModuleMetaData.imports) {
+    testModuleMetaData.imports = []
+  }
+
+  testModuleMetaData.declarations.push(component)
+
+  if (!testModuleMetaData.imports.includes(CommonModule)) {
+    testModuleMetaData.imports.push(CommonModule)
+  }
+
+  return testModuleMetaData
+}
+
+/**
+ * Initializes the TestBed
+ *
+ * @param {Type<T>} component Angular component being mounted
+ * @param {TestBedConfig} config TestBed configuration passed into the mount function
+ * @returns {TestBed} TestBed
+ */
+function initTestBed<T extends object> (
+  component: Type<T>,
+  config: TestBedConfig<T>,
+): TestBed {
+  const { providers, ...configRest } = config
+
+  const testBed: TestBed = getTestBed()
+
+  testBed.resetTestEnvironment()
+
+  testBed.initTestEnvironment(
+    BrowserDynamicTestingModule,
+    platformBrowserDynamicTesting(),
+    {
+      teardown: { destroyAfterEach: false },
+    },
+  )
+
+  testBed.configureTestingModule({
+    ...bootstrapModule(component, configRest),
+  })
+
+  if (providers != null) {
+    testBed.overrideComponent(component, {
+      add: {
+        providers,
+      },
+    })
+  }
+
+  return testBed
+}
+
+/**
+ * Creates the ComponentFixture
+ *
+ * @param component Angular component being mounted
+ * @param testBed TestBed
+ * @param autoDetectChanges boolean flag defaulted to true that turns on change detection automatically
+ * @returns {ComponentFixture<T>} ComponentFixture
+ */
+function setupFixture<T extends object> (
+  component: Type<T>,
+  testBed: TestBed,
+  autoDetectChanges: boolean,
+): ComponentFixture<T> {
+  const fixture = testBed.createComponent(component)
+
+  fixture.whenStable().then(() => {
+    fixture.autoDetectChanges(autoDetectChanges)
+  })
+
+  return fixture
+}
+
+/**
+ * Gets the componentInstance and Object.assigns any inputs() passed in the TestBedConfig
+ *
+ * @param {TestBedConfig} config TestBed configuration passed into the mount function
+ * @param {ComponentFixture<T>} fixture Fixture for debugging and testing a component.
+ * @returns {T} Component being mounted
+ */
+function setupComponent<T extends object> (
+  config: TestBedConfig<T>,
+  fixture: ComponentFixture<T>,
+): T {
+  let component: T = fixture.componentInstance
+
+  if (config?.inputs) {
+    component = Object.assign(component, config.inputs)
+  }
+
+  return component
+}
+
```

Comment:
How do we actually mount the component? React/Vue use a `data-cy-root` selector which they import https://github.com/cypress-io/cypress/blob/develop/npm/react/src/mount.ts#L8

I don't see how we go from component -> mount to DOM here. Am I missing it, or is that implemented in another PR?

https://github.com/cypress-io/cypress/pull/22858/files#diff-617e9faa6b88ce75b6ca3135c54ae7000965719cc2d6c9ae9813178d9da19170

Reply:
By default Angular mounts the component for you to the DOM using the `angular.json` build.options.index property. In [THIS PR](https://github.com/cypress-io/cypress/pull/22314) we change this value from what the user has to `cypress/support/component-index.html`. It is then in that file (that we will need to generate as part of the migration work) that angular will mount the component